lisa/operating_system.py

```diff
@@ -150,15 +196,35 @@ def type_name(cls) -> str:
     def name_pattern(cls) -> Pattern[str]:
         return re.compile(f"^{cls.type_name()}$")
 
-    def _install_packages(self, packages: Union[List[str]]) -> None:
+    def _install_packages(
+        self, packages: Union[List[str]], signed: bool = True
+    ) -> None:
+        raise NotImplementedError()
+
+    def _query_package(self, package: str, signed: bool = True) -> bool:
         raise NotImplementedError()
 
     def _initialize_package_installation(self) -> None:
         # sub os can override it, but it's optional
         pass
 
+    # _install_package_from_url
+    # Used if the package to be installed needs to be
+    # downloaded from a url first.
+    def _install_package_from_url(
+        self,
+        package: str,
+        signed: bool = True,
+    ) -> None:
+        # when package is URL, download the package first at the working path.
+        wget_tool = self._node.tools[Wget]
+        pkg = wget_tool.get(package, str(self._node.working_path))
+        self.install_packages(f"{pkg}", signed)
```

Comment:
Should it be a method called `install_package_from_file`?

Reply:
the `install_packages` are installed from names, that's major scenario, that we install packages from names in package repo. I'm not sure if all package tools can support package names and file names together. If it's supported by all package tools, we can call it like current implementation.

Reply:
yes they install from names, but ultimately the name represent some file like *.rpm / *.deb in the tools's repositories.
So basically it maps the name to a file.

Reply:
ok, if it works, we can do like this way.